### PR TITLE
Add tests for resolve packages built from different dist versions

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/34650

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_e/resources/project-info.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+org="samjs"
+name="package_e"
+version="0.1.0"
```
Comment (azinneera) [inline project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_e/resources/project-info.properties:3]: Missing new line

File: project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_c/modules/mod_c2/mod2.bal [hunk 2]
```diff
@@ -0,0 +1,2 @@
+public function func2() {
+}
```
Comment (azinneera) [inline project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_c/modules/mod_c2/mod2.bal:2]: Missing new line

File: project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_e/resources/project-info.properties (hunk 1 above)
Comment (pramodya1994) [reply project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_e/resources/project-info.properties]: Added

File: project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_c/modules/mod_c2/mod2.bal (hunk 2 above)
Comment (pramodya1994) [reply project-api/project-api-test/src/test/resources/projects_for_resolution_tests/packages_for_various_dist_test/package_c/modules/mod_c2/mod2.bal]: Added

**Review by azinneera — APPROVED** [review]

**Pull request merged**